evm

```diff
@@ -118,23 +118,79 @@ function evm_system() {
   export PATH
 }
 
-function _unzip-strip() {
+function _unzip_strip() {
   # shopt is a bash built-in command. Zsh, has setopt instead
   # so, set the command according to the type of shell (only supports bash and zsh)
   local SHOPT="shopt"
-  if [[ "$EVM_PS_SHELL" == *zsh ]]; then
+  if [[ "$EVM_PS_SHELL" == *zsh ]] || [[  "$(which $SHELL)" == *"/zsh" ]] ; then
     SHOPT="setopt"
   fi
 
   local zip="$1"
   local dest="${2:-.}"
-  local temp=$(mktemp -d) && unzip -d "$temp" "$zip" && mkdir -p "$dest" && $SHOPT -s dotglob && local f=("$temp"/*) &&
+  local -i _err=0
+  local temp=$(mktemp -d)
+  _err=$?
+  if [ ${_err} -ne 0 ] ; then
+  {
```

Comment:
Please remove all those curly brackets from the body of the if statements, there are not necessary.

Reply:
Curly brackets I always add for readability, but sure I can remove

Reply:
Readability is important but a subjective matter. Curly brackets have specific meaning in bashscript and should only be used for what they are meant for. If the goal is to make the body of the `if` statement explicitly visible, you can always move the `then` keyword to the next line:

```
if condition
then
   logic
fi
``` 

Which I particularly am not a big fan either, but at least you would be using the language constructions correctly.

Reply:
> Please remove all those curly brackets from the body of the if statements, there are not necessary.

removed

Reply:
> Readability is important but a subjective matter. Curly brackets have specific meaning in bashscript and should only be used for what they are meant for. If the goal is to make the body of the `if` statement explicitly visible, you can always move the `then` keyword to the next line:
> 
> ```
> if condition
> then
>    logic
> fi
> ```
> 
> Which I particularly am not a big fan either, but at least you would be using the language constructions correctly.

I actually think that curly brackets makes it more readable